src/index.ts

```diff
@@ -0,0 +1,52 @@
+import { resolve, join } from 'path';
+import { readJson } from 'fs-extra';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import remoteTags from 'remote-git-tags';
+
+const REQUIRED_TAG_FORMAT = '$version';
+
+// eslint-disable-next-line camelcase
+const cloneUrl: string = github.context.payload.repository?.clone_url;
+const tagFormat: string = core.getInput('tag-format');
+
+async function getVersion(): Promise<string> {
+  const packagePath = join(core.getInput('package-directory'), 'package.json');
+  const { version } = await readJson(resolve(packagePath));
+
+  if (!version) {
+    throw new Error(`Missing "version" in ${packagePath}`);
+  }
+
+  return version;
+}
+
+async function run(): Promise<void> {
```

Comment:
Don't need to for now, but in the future I would be open to requiring jsdoc comment with [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc)

Reply:
Just to enforce comments in our code or for any other purpose with Typescript?

Reply:
Just to enforce the comments, also then you get that description in the autocomplete/hover of each function (both in js and ts)

Reply:
Added